### PR TITLE
Fixes issue where compile errors are not caught during build steps

### PR DESCRIPTION
**Pull request by wbreza**
- Fixes missing import statements in MockFactory
- Adds new CI step to confirm TypeScript compile succeeds before running tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,17 +28,23 @@ install:
 
 jobs:
   include:
+    - stage: compile
+      name: "TypeScript compiler for Node"
+      node_js:
```
Comment (mydiemho) [inline .travis.yml]: this is not running `node 10` anymore

File: .travis.yml [hunk 2]
```diff
@@ -26,17 +28,23 @@ install:
 
 jobs:
   include:
+    - stage: compile
+      name: "TypeScript compiler for Node"
+      node_js: 
+        - "8"
+        - "10"
+      script: 
+        - npm run compile
```
Comment (mydiemho) [inline .travis.yml]: do you just want to compile before running unit tests?

File: .travis.yml (hunk 1 above)
Comment (wbreza) [reply .travis.yml]: What is the correct syntax to run on both?

Comment (wbreza) [reply .travis.yml]: Figured it out - reverted most of what i had and just added the compile in the existing stage.

**Comment by wbreza** [comment]
@mydiemho Please take a second look. Thanks!

File: package.json [hunk 3]
```diff
@@ -14,8 +14,9 @@
     "test": "jest",
     "test:ci": "npm run test -- --ci",
     "test:coverage": "npm run test -- --coverage",
+    "compile": "tsc",
     "prebuild": "shx rm -rf lib/ && npm run test",
-    "build": "tsc"
+    "build": "npm run compile"
```
Comment (mydiemho) [inline package.json]: is this just for backward compatibility?  seem redundant

**Pull request merged**